File: Python/Binary Trees/L_2SumBST.py
def findTarget(root, k: int) -> bool:
        visit = set()
        def dfs(node):
            if not node:
                return False
            if k - node.val in visit:
                return True
            else:
                visit.add(node.val)
            return dfs(node.left) or dfs(node.right)
        return dfs(root)

File: cpp/Binary Tree/L_2SumBST.cpp
#include<bits/stdc++.h>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

bool findTarget(TreeNode* root, int k) {
    unordered_set <int> visit;
    return dfs(root, k, visit);
}

bool dfs(TreeNode* node, int k, unordered_set <int> &visit) {
    if (!node) return false;
    if (visit.find(k - node->val) != visit.end()) return true;
    else visit.insert(node->val);
    return dfs(node->left, k, visit) or dfs(node->right, k, visit);
}